### ISSUE TITLE
feature_statistics_path

### ISSUE THREAD

**Issue by LRLLRL** [ZYYSzj/Selective-Joint-Fine-tuning#4]
feature_statistics_path: "examples/cvpr2017/mit67/image_retrieval/feature_extraction/mit67_alexnet_color_conv11_5_feature_stats_domain.txt"
I don't understand this address.Could you explain?

**Comment by ZYYSny** [comment]
It stands for the bounds of the soft histogram bins. Before extracting
features，you should run g

LRLLRL <notifications@github.com>于2018年3月13日 周二下午6:12写道：

**Comment by ZYYSny** [comment]
It stands for the bounds of the soft histogram bins. Before extracting
features，you should run get_feature_stats.sh twice. The first time is to
get the feature value range of each channel. The second time is to get the
soft bins of histograms.

戈维峰 <weifeng.ge.ic@gmail.com>于2018年3月14日 周三上午8:24写道：

**Comment by daa233** [comment]
Hi, @ZYYSzj, when running `get_feature_stats.sh` the first time, how to designate the `previous_statistics_path` in `image_retrieval/feature_stats/stats_deploy.prototxt`? Do we have one txt file for that parameter before we run it for the first time?

I am confused because I get an error of `Check failed: idx == total_channels_ (0 vs. 1376)` when run the script for the first time. Thanks for any help.

**Comment by 17860846552** [comment]
@DAA233  I got the same problem as you, have you solved this problem?

**Comment by daa233** [comment]
@17860846552 No, I haven't... 😢  Not very clear about some details in `xxx_deploy.prototxt`.

**Comment by ZYYSny** [comment]
@LRLLRL  @DAA233  @17860846552 I have updated the ReadMe.md in the get_the_features_statistics branch. I am busy in other projects now. Sorry for the delay.

**Comment by daa233** [comment]
@ZYYSzj Thanks for updating.
I think I have gotten the main idea. When we run the code for the first time, we need to provide fake files like the format of "caltech_alexnet_color_feature_stats.txt". Otherwise, we will get the above error `Check failed: idx == total_channels_ (0 vs. 1376)` since we don't provide the fake files. And before running the code for the second time, we should rename "caltech_alexnet_color_feature_stats.txt" as "caltech_alexnet_color_feature_stats_previous.txt".

**Comment by ZYYSny** [comment]
@DAA233 You are right. Since the codes is not well written. You need to provide fake files in the same format above. If you have 1376 feature maps, you need randomly generate 1376 lines. Then you can run the script again. You just need to run over you training set one epoch when you run the scripts.

**Comment by 17860846552** [comment]
@ZYYSzj  It really works!  Thank you for taking time out of your busy schedule.
